lua/nvim-tree/actions/init.lua

```diff
@@ -126,24 +127,34 @@ local function merge_mappings(user_mappings)
     return M.mappings
   end
 
+  local function is_empty(s)
+    return s == ''
+  end
+
   local user_keys = {}
   local removed_keys = {}
+  -- remove default mappings if action is a empty string
   for _, map in pairs(user_mappings) do
     if type(map.key) == "table" then
       for _, key in pairs(map.key) do
         table.insert(user_keys, key)
-        if not map.action then
+        if is_empty(map.action) then
           table.insert(removed_keys, key)
         end
       end
     else
       table.insert(user_keys, map.key)
-      if not map.action then
+      if is_empty(map.action) then
         table.insert(removed_keys, map.key)
       end
     end
+
     if map.action and type(map.action_cb) == "function" then
-      M.custom_keypress_funcs[map.action] = map.action_cb
+      if not is_empty(map.action) then
+        M.custom_keypress_funcs[map.action] = map.action_cb
```

Comment:
here it will try to set [nil] = map.action_cb if the user forgets to pass in map.action and is nil

Reply:
You forgot the previous condition `if mal.action and...`.